plugins/digitalocean/personal_access_token.go

```diff
@@ -32,6 +36,42 @@ func PersonalAccessToken() schema.CredentialType {
 		DefaultProvisioner: provision.EnvVars(map[string]sdk.FieldName{
 			"DIGITALOCEAN_ACCESS_TOKEN": fieldname.Token,
 		}),
-		Importer: importer.TryAllEnvVars(fieldname.Token, "DIGITALOCEAN_ACCESS_TOKEN"),
+		Importer: importer.TryAll(
+			importer.TryAllEnvVars(fieldname.Token, "DIGITALOCEAN_ACCESS_TOKEN"),
+			importer.MacOnly(TryDigitalOceanConfigFile("~/Library/Application Support/doctl/config.yaml")),
+			importer.LinuxOnly(TryDigitalOceanConfigFile(digitalOceanConfigFileOnLinux())),
+		),
 	}
 }
+
+func TryDigitalOceanConfigFile(path string) sdk.Importer {
+	return importer.TryFile(path, func(ctx context.Context, contents importer.FileContents, in sdk.ImportInput, out *sdk.ImportAttempt) {
+		var config Config
+		if err := contents.ToYAML(&config); err != nil {
+			out.AddError(err)
+			return
+		}
+
+		if config.AccessToken == "" {
+			return
+		}
+
+		out.AddCandidate(sdk.ImportCandidate{
+			Fields: map[sdk.FieldName]string{
+				fieldname.Token: config.AccessToken,
+			},
+		})
+	})
+}
+
+func digitalOceanConfigFileOnLinux() string {
+	basePath := "~/.config"
+	if xdg := os.Getenv("XDG_CONFIG_HOME"); xdg != "" {
+		basePath = xdg
+	}
+	return path.Join(basePath, "doctl/config.yaml")
+}
```

Comment:
This is not a DigitalOcean-only thing. This goes for (I think) every plugin we have that uses `~/.config`, which are quite a lot.

A while ago I [experimented](https://github.com/1Password/shell-plugins/commit/d44a749ba7d89e729a7a11366af6c3ac67cfd943) with a way to standardize this pattern that could be interesting to further explore.

Reply:
You're right that we should seek to standardise this. I opened an issue for this, I propose we tackle it outside this MR.

https://github.com/1Password/shell-plugins/issues/153